kotlin/authenticate-with-oauth/src/main/java/com/esri/arcgisruntime/sample/authenticatewithoauth/MainActivity.kt

```diff
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.authenticatewithoauth
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.security.AuthenticationManager
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler
+import com.esri.arcgisruntime.security.OAuthConfiguration
+import kotlinx.android.synthetic.main.activity_main.*
+import java.net.MalformedURLException
+
+
+class MainActivity : AppCompatActivity() {
+
+  companion object {
+    private val TAG: String = MainActivity::class.java.simpleName
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    try {
+      // set up an oauth config with url to portal, a client id and a re-direct url
+      // a custom client id for your app can be set on the ArcGIS for Developers dashboard under
+      // Authentication --> Redirect URIs
+      val oAuthConfiguration = OAuthConfiguration(
+        getString(R.string.portal_url),
+        getString(R.string.oauth_client_id),
+        getString(R.string.oauth_redirect_uri) + "://" + getString(R.string.oauth_redirect_host)
+      )
+
+      // setup AuthenticationManager to handle auth challenges
+      val defaultAuthenticationChallengeHandler = DefaultAuthenticationChallengeHandler(
+        this
+      )
+
+      // use the DefaultChallengeHandler to handle authentication challenges
+      AuthenticationManager.setAuthenticationChallengeHandler(defaultAuthenticationChallengeHandler)
+
+      // add an OAuth configuration
+      // NOTE: you must add the DefaultOAuthIntentReceiver Activity to the app's manifest to handle starting a browser
+      AuthenticationManager.addOAuthConfiguration(oAuthConfiguration)
+
+      // load the portal and add the portal item as a map to the map view
+      val portal = Portal(getString(R.string.portal_url))
+      val portalItem = PortalItem(portal, getString(R.string.webmap_world_traffic_id))
+      mapView.apply {
+        map = ArcGISMap(portalItem)
+      }
+    } catch (e: MalformedURLException) {
+      logError("Error in OAuthConfiguration URL: " + e.message)
+    }
+  }
+
+  override fun onResume() {
+    super.onResume()
+    mapView.resume()
+  }
+
+  override fun onPause() {
+    // normally, you won't want to clear credentials once a device has been verified. These calls are made to keep this
+    // sample from interfering with other authentication samples
+    AuthenticationManager.CredentialCache.clear();
```

Comment:
can lose the ; on this and the next line

It was my idea to put this in OnPause in the original Java sample, but I wonder if it might more usefully be placed at the start of onCreate--This would reset credentials each time the app is run without it being reset if someone just loses focus of the app